### ISSUE TITLE
IEnumerable<string> does not respect TypeMap

### ISSUE THREAD

**Issue by aaronasmith** [StackExchange/Dapper#200]
Setting the typemap to map string to AnsiString works with a single parameter.  When passing in an IEnumerable<string> it automatically uses String instead of looking up the DBType in the typemap.

**Comment by NickCraver** [comment]
Is this a current issue? There have been a lot of generic changes since this was opened (and fell through the cracks) - sorry about that!

**Comment by aaronasmith** [comment]
Yes, I just checked and with the latest version from NuGet this is still an issue.
```c#
SqlMapper.AddTypeMap(typeof(string), DbType.AnsiString);
var results = connection.Query("Select * from MyTable where MyValue in @thevalues", new { thevalues= new List<DbString> { new List<string> {  "a value", "another value" } }).ToList();
```
This uses nvarchar instead of varchar.  It's easy enough to get around by using DbString.
